stellargraph/core/graph.py

```diff
@@ -41,166 +41,232 @@
-            described above. **Warning**: this is experimental (see warning above for more details).
+            and they have to be unique across all types.
 
-        source_column: str, optional
+        is_directed (bool, optional):
+            If True, the data represents a directed multigraph, otherwise an undirected multigraph.
+
+        source_column (str, optional):
             The name of the column to use as the source node of edges in the ``edges`` edge list
             argument.
 
-        target_column: str, optional
+        target_column (str, optional):
             The name of the column to use as the target node of edges in the ``edges`` edge list
             argument.
 
-        edge_weight_label: str, optional
-            The name of the attribute to use as the weight of edges (for the `nodes`/`edges`
-            DataFrame parameters, this is the name of the column to use).
+        edge_weight_column (str, optional):
+            The name of the column in each of the ``edges`` DataFrames to use as the weight of
+            edges. If the column does not exist in any of them, it is defaulted to ``1``.
+
+        node_type_default (str, optional):
+            The default node type to use, if ``nodes`` is passed as a DataFrame (not a ``dict``).
+
+        edge_type_default (str, optional):
+            The default edge type to use, if ``edges`` is passed as a DataFrame (not a ``dict``).
+
+        dtype (numpy data-type, optional):
+            The numpy data-type to use for the features extracted from each of the ``nodes`` DataFrames.
+
+        graph:
+            Deprecated, use :meth:`from_networkx`.
+        node_type_name:
+            Deprecated, use :meth:`from_networkx`.
+        edge_type_name:
+            Deprecated, use :meth:`from_networkx`.
+        feature_name:
+            Deprecated, use :meth:`from_networkx`.
+        target_name:
+            Deprecated, use :meth:`from_networkx`.
+        edge_weight_label:
+            Deprecated, use :meth:`from_networkx`.
+        node_features:
+            Deprecated, use :meth:`from_networkx`.
     """
 
     def __init__(
         self,
-        graph=None,
+        nodes=None,
+        edges=None,
+        *,
         is_directed=False,
-        edge_weight_label="weight",
-        node_type_name=globalvar.TYPE_ATTR_NAME,
-        edge_type_name=globalvar.TYPE_ATTR_NAME,
+        source_column=globalvar.SOURCE,
+        target_column=globalvar.TARGET,
+        edge_weight_column=globalvar.WEIGHT,
         node_type_default=globalvar.NODE_TYPE_DEFAULT,
         edge_type_default=globalvar.EDGE_TYPE_DEFAULT,
+        dtype="float32",
+        # legacy arguments:
+        graph=None,
+        node_type_name=globalvar.TYPE_ATTR_NAME,
+        edge_type_name=globalvar.TYPE_ATTR_NAME,
         feature_name=globalvar.FEATURE_ATTR_NAME,
         target_name=globalvar.TARGET_ATTR_NAME,
```

Comment:
were these always unused?

Reply:
Haha thanks for tracking down the history, yeah it sounds like we could just remove them here

Reply:
Done.